It broke for browserify

Browserify does not check the existence of the "browser-field" in the npm package.json. Even with babelify, it does not work.

While testing with open-flight-planner got some interesting findings:
* browserify does not like `"browser": "dist/localforage.js"`
  * it gives: `Error: Cannot find module './../utils/serializer'`
* it works fine with `"browser": "build/es5src/localforage.js",`
  * if it is included of course. The whole point of this is to leave all the project-specific babel configurations away from our users

Seems like we need some better/new browserify tests.



- tofumatt

On Mon, Aug 3, 2015 at 6:23 PM, Thodoris Greasidis
<notifications@github.com> wrote:

> While testing with open-flight-planner got some interesting findings:
> * browserify does not like `"browser": "dist/localforage.js"`
>   * it gives: `Error: Cannot find module './../utils/serializer'`
> * it works fine with `"browser": "build/es5src/localforage.js",`
>   * if it is included of course. The whole point of this is to leave all the project-specific babel configurations away from our users
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/mozilla/localForage/issues/412#issuecomment-127336708

--- a/.babelrc
+++ b/.babelrc
@@ -1,0 +1,4 @@
+{
+  "loose": "all",
+  "plugins": ["system-import-transformer"]
+}

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -34,12 +34,9 @@ module.exports = exports = function(grunt) {
     grunt.initConfig({
         babel: {
             options: {
-                loose: 'all',
                 modules: 'umd',
                 moduleIds: true,
-                // sourceMap: true,
-                getModuleId: babelModuleIdProvider,
-                plugins: ['system-import-transformer']
+                getModuleId: babelModuleIdProvider
             },
             dist: {
                 files: {
@@ -55,10 +52,8 @@ module.exports = exports = function(grunt) {
             client: {
                 options: {
                     transform: [['babelify', {
-                        loose: 'all',
                         moduleIds: true,
-                        getModuleId: babelModuleIdProvider,
-                        plugins: ['system-import-transformer']
+                        getModuleId: babelModuleIdProvider
                     }]]
                 },
                 src: [

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -49,18 +49,8 @@ module.exports = exports = function(grunt) {
             }
         },
         browserify: {
-            client: {
-                options: {
-                    transform: [['babelify', {
-                        moduleIds: true,
-                        getModuleId: babelModuleIdProvider
-                    }]]
-                },
-                src: [
-                    'bower_components/es6-promise/promise.js',
-                    'src/**/*.js',
-                    'test/runner.browserify.js'
-                ],
+            package_bundling_test: {
+                src: 'test/runner.browserify.js',
                 dest: 'test/localforage.browserify.js'
             }
         },
@@ -211,7 +201,7 @@ module.exports = exports = function(grunt) {
                     'test/runner.js',
                     'test/test.*.*'
                 ],
-                tasks: ['jshint', 'jscs', 'shell:component', 'browserify', 'mocha:unit']
+                tasks: ['jshint', 'jscs', 'shell:component', 'browserify:package_bundling_test', 'mocha:unit']
             }
         }
     });
@@ -230,7 +220,7 @@ module.exports = exports = function(grunt) {
         'jshint',
         'jscs',
         'shell:component',
-        'browserify',
+        'browserify:package_bundling_test',
         'connect:test',
         'mocha'
     ];

--- a/package.json
+++ b/package.json
@@ -48,5 +48,8 @@
   },
   "dependencies": {
     "promise": "^5.0.0"
+  },
+  "browserify": {
+    "transform": ["babelify"]
   }
 }

--- a/test/runner.browserify.js
+++ b/test/runner.browserify.js
@@ -1,1 +1,5 @@
-window.localforage = require('./../dist/localforage');
+if (typeof Promise === 'undefined') {
+    require('../bower_components/es6-promise/promise');
+}
+// require localforage as defined in package.json
+window.localforage = require('../');

